### DISCUSSION TITLE
Custom hooks during PR creation

### DISCUSSION THREAD

**Discussion by vblagoje** [community/community#76937] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

Is there a way to invoke a Github programmatic hook whenever a new PR request via Web Browser is triggered? I want to generate PR text automatically using LLMs.

**Comment by kdinthehouse** [comment, marked as answer]
Yes you can

To do this you will need to create a GitHub App and register a webhook for the "pull request" event. The webhook URL will point to a server that you control, where you will have a script that runs when the webhook is triggered. The script can then use an LLM to generate the PR text and post it to GitHub.

Steps

Create a GitHub App: You will need to create a GitHub App in order to register a webhook. You can do this from the GitHub Developer Settings page.

Register a webhook for the "pull request" event: Once you have created a GitHub App, you can register a webhook for the "pull request" event. This will tell GitHub to send a webhook notification to your server whenever a new pull request is created.

Set up a server to receive webhooks: You will need to set up a server that can receive webhook notifications from GitHub. This could be a server that you host yourself, or you could use a cloud-based serverless platform like AWS Lambda or Azure Functions.

Write a script to generate PR text: You will need to write a script that runs when the webhook is triggered. This script can use an LLM to generate the PR text and post it to GitHub.
